indico/modules/events/editing/controllers/backend/timeline.py

```diff
@@ -388,7 +388,11 @@ def _process(self):
                 folder_name = secure_filename(file_type.name, f'file-type-{file_type.id}')
 
                 with file.storage.get_local_path(file.storage_file_id) as filepath:
-                    zip_handler.write(filepath, os.path.join(folder_name, filename))
+                    info = ZipInfo.from_file(filepath)
+                    info.filename = os.path.join(folder_name, filename)
+                    info.date_time = self.revision.created_dt.astimezone(self.event.tzinfo).timetuple()[:6]
+                    with open(filepath, 'rb') as fd:
+                        zip_handler.writestr(info, fd.read())
```

Comment:
Isn't this code (based on the `zip_handler.write()` code better to avoid reading the whole file into memory?

            with open(filename, 'rb') as src, zip_handler.open(info, 'w') as dest:
                shutil.copyfileobj(src, dest, 1024*8)

In fact you can probably use `.open()` instaed of `.get_local_path()` to get the `src` file object